plugins/scanners/scanoss/src/main/kotlin/ScanOss.kt

```diff
@@ -99,4 +108,71 @@ class ScanOss(
         val endTime = Instant.now()
         return generateSummary(startTime, endTime, results)
     }
+
+    data class ProcessedRules(
+        val includeRules: List<Rule>,
+        val ignoreRules: List<Rule>,
+        val replaceRules: List<ReplaceRule>,
+        val removeRules: List<RemoveRule>
+    )
+
+    private fun buildSettingsFromORTContext(context: ScanContext): ScanossSettings {
+        val rules = processSnippetChoices(context.snippetChoices)
+        val bom = Bom.builder()
+            .ignore(rules.ignoreRules)
+            .include(rules.includeRules)
+            .replace(rules.replaceRules)
+            .remove(rules.removeRules)
+            .build()
+        return ScanossSettings.builder().bom(bom).build()
+    }
+
+    fun processSnippetChoices(snippetChoices: List<SnippetChoices>): ProcessedRules {
+        val includeRules = mutableListOf<Rule>()
+        val ignoreRules = mutableListOf<Rule>()
+        val replaceRules = mutableListOf<ReplaceRule>()
+        val removeRules = mutableListOf<RemoveRule>()
+
+        snippetChoices.forEach { snippetChoice ->
+            snippetChoice.choices.forEach { choice ->
+                when (choice.choice.reason) {
+                    SnippetChoiceReason.ORIGINAL_FINDING -> {
+                        includeRules.addFinding(choice)
+                    }
+
+                    SnippetChoiceReason.NO_RELEVANT_FINDING -> {
+                        removeRules.removeFinding(choice)
+                    }
+
+                    SnippetChoiceReason.OTHER -> {
+                        logger.info {
+                            "Encountered OTHER reason for snippet choice in file ${choice.given.sourceLocation.path}"
+                        }
+                    }
+                }
+            }
+        }
+
+        return ProcessedRules(includeRules, ignoreRules, replaceRules, removeRules)
+    }
+
+    private fun MutableList<Rule>.addFinding(choice: SnippetChoice) {
+        this += Rule.builder()
+            .purl(choice.choice.purl)
+            .path(choice.given.sourceLocation.path)
+            .build()
+    }
+
+    private fun MutableList<RemoveRule>.removeFinding(choice: SnippetChoice) {
+        this += RemoveRule.builder().apply {
+            path(choice.given.sourceLocation.path)
+
+            // Set line range only if both `sourceLocation.startLine` and `sourceLocation.endLine` are positive numbers.
+            // If either line is zero or negative, the rule will apply to the entire file.
+            if (choice.given.sourceLocation.startLine > 0 && choice.given.sourceLocation.endLine > 0) {
```

Comment:
As the `TextLocation` constructor limits valid line numbers, see

https://github.com/oss-review-toolkit/ort/blob/6e68575eefb693307170d9b7af978260fa1b80bb/model/src/main/kotlin/TextLocation.kt#L58-L60

I believe this check can be simplified to just `choice.given.sourceLocation.startLine != TextLocation.UNKNOWN_LINE`.

Reply:
Note that if changing the code, the comment should be slightly adjusted as well.

Reply:
Replaced hardcoded `choice.given.sourceLocation.startLine > 0` checks with UNKNOWN_LINE constants for better semantics. 

I kept both checks for readability, but I'm happy to simplify to a single check if you prefer that approach

Reply:
> I kept both checks for readability, but I'm happy to simplify to a single check if you prefer that approach

I wasn't super happy with the single check either, as it reads a bit if don't know the internals. I was also thinking about introducing [this](https://github.com/oss-review-toolkit/ort/pull/10280/files#diff-f9e13990c51fa75dc77d2c7898628752db57f0fc7addb2e368850ad452d042ebR71). How about that?

Reply:
I was thinking also to introduce a method like you did on #10280. Is that PR going to be merge today?

Reply:
> Is that PR going to be merge today?

I'm confident at it will.